sylt/src/formatter.rs

```diff
@@ -0,0 +1,613 @@
+                write_type(dest, indent, ty)?;
+                if kind.immutable() {
+                    write!(dest, " : ")?;
+                } else {
+                    write!(dest, " = ")?;
+                }
+            }
+            write_expression(dest, indent, value)?;
+        }
+        StatementKind::EmptyStatement => (),
+        StatementKind::If {
+            condition,
+            pass,
+            fail,
+        } => {
+            if matches!(fail.kind, StatementKind::EmptyStatement) {
+                for comment in &fail.comments {
+                    write!(dest, "// {}\n", comment)?;
+                    write_indents(dest, indent)?;
+                }
+            }
+
+            write!(dest, "if ")?;
+            write_expression(dest, indent, condition)?;
+            write!(dest, " ")?;
+            write_statement(dest, indent, pass)?;
+            if !matches!(fail.kind, StatementKind::EmptyStatement) {
+                write!(dest, " else ")?;
+                write_statement(dest, indent, fail)?;
+            }
+        }
+        StatementKind::IsCheck { lhs, rhs } => {
+            write!(dest, ":")?;
+            write_type(dest, indent, lhs)?;
+            write!(dest, " is :")?;
+            write_type(dest, indent, rhs)?;
+        }
+        StatementKind::Loop { condition, body } => {
+            write!(dest, "loop ")?;
+            write_expression(dest, indent, condition)?;
+            write!(dest, " ")?;
+            write_statement(dest, indent, body)?;
+        }
+        StatementKind::Ret { value } => {
+            write!(dest, "ret ")?;
+            write_expression(dest, indent, value)?;
+        }
+        StatementKind::StatementExpression { value } => write_expression(dest, indent, value)?,
+        StatementKind::Unreachable => {
+            write!(dest, "<!>")?;
+        }
+        StatementKind::Use {
+            path,
+            name,
+            file: _,
+        } => {
+            write!(dest, "use ")?;
+            write_identifier(dest, path)?;
+            if let NameIdentifier::Alias(alias) = name {
+                write!(dest, " as ")?;
+                write_identifier(dest, alias)?;
+            }
+        }
+    }
+
+    Ok(())
+}
+
+/// Replace consecutive empty statements with one empty statement with all comments of the previous statements.
+//TODO(gu): Rewrite the formatter to use moves instead of borrows. Then we wouldn't need to clone when passing
+//          into this function.
+fn merge_empty_statements(mut statements: Vec<Statement>) -> Vec<Statement> {
+    // Reverse since
+    // - we always want to remove and look at the first statement and
+    // - pop() is faster than remove(0).
+    statements.reverse();
+
+    let mut ret = Vec::new();
+    while let Some(mut statement) = statements.pop() {
+        if !matches!(statement.kind, StatementKind::EmptyStatement) {
+            ret.push(statement);
+            continue;
+        }
```

Comment:
Can't you just remove this block? The while loop below does the same check, no? I haven't tried it though.